src/libcore/slice.rs

```diff
@@ -1183,6 +1183,34 @@ impl<'a, T> Iterator for Windows<'a, T> {
             (size, Some(size))
         }
     }
+
+    #[inline]
+    fn count(self) -> usize {
+        self.size_hint().0
+    }
+
+    #[inline]
+    fn nth(&mut self, n: usize) -> Option<Self::Item> {
+        let end = self.size + n;
```

Comment:
Not an obstacle! There are several ways to do this, you can use .checked_add for example, but i thought overflowing_add would be the correct one for this particular line.

(saturating_add produces usize::MAX instead of overflowing, but that's not correct for the comparison `end > self.len()` — usize::MAX is not guaranteed to be larger).